docs/algorand/how-it-works/data-availability.mdx

```diff
@@ -0,0 +1,162 @@
+
+If a single one of the transactions are invalid/fail, then the batch is ignored because it is an invalid state transition. Even though the batch was verified to be ready to be added to the sequenced rollup history and the data was made available, the actual contents of the batch were invalid, and as such it is ignored.
+
+In either case, whether the batch is applied and the rollup progresses forward, or if the batch is ignored leaving the rollup state unchanged, we move back into the batch proposal phase where a new batch has the potential to be added to the rollup history.
+
+Of note, all of this logic is encoded off-chain in the rollup node/auxillary software. This defined protocol is how all the rollup nodes update the state of the rollup itself locally. Anyone can spin up their own rollup node and re-read the Algorand history to deterministically arrive at the same state.
+
+## Advanced Merkelized Batch Proposal Approach
+
+The basic merkelized batch proposal approach allows us to get past the 16kb limit, however only works in a context where there is a single batcher (or fully trusted set of batchers going in order). In this advanced variant, we will be upgrading the batch proposal phase with some more complex dynamics to allow for an open set of batchers to exist who may be adversarial and try to cause problems.
+
+In this advanced approach, we will be providing all whitelisted batchers (which will likely be a subset of the bridge validators, who eventually will be required to stake assets) the opportunity to be selected as the batch submitter to extend the rollup history.
+
+
+After the previous batch has either been finalized or ignored (if invalid), we return back to the batch proposal phase. Any of the batchers can now propose a new batch by posting the merkle root of it on-chain. Once a batch proposal is posted by one batcher, all other batchers have 2 blocks to submit their own batch proposals.
+
+After the 2 blocks have passed, one of the batch proposals is pseudo-randomly selected via using the hash of the previous latest batch as a seed to deterministically choose one. The submitter of the batch proposal now is required to post the data of the batch as the protocol moves into the batch availability phase.
+
+The batch availability phase proceeds just as expected, wherein the selected batch submitter must post all of the groupings + merkle proofs of their batch in the next 6 blocks. If they fail to do this (or if during batch validation the batch is invalid), the batch is ignored and the protocol returns to the batch proposal phase.
+
+Depending on the complexity of the implementation, if specific batchers repeatedly keep failing to submit the data of batches they submit, it would be wise to stake slash them for attempting to DoS the system or simply failing to ensure their software is working properly.
```

Comment:
```suggestion
Depending on the complexity of the implementation, if specific batchers repeatedly keep failing to submit the data of batches they submit, it would be wise to stake slash them for attempting to DoS the system or for simply failing to ensure their software is working properly.
```